ls1intum/Artemis
# Show result in exam summary page after result publication date

### Checklist

### Motivation and Context


Results and feedback for the exam should be shown on the summary page of the exam.

### Description

- Show the result after the publishResultDate of the exam and a button that leads to the feedback (with text & modeling)
- Display Quizzes directly in the summary
- Show programming exercises in the summary
- Forbid students to access the submission of the exercise by navigating to it using the url before the publish date

### Steps for Testing


1. Take part in an exam.
2. Assess the exam.
3. Set publishResultDate on the exam.
4. Look for the results and assessments on the result summary page (last page of the exam itself) and check for missing and/or wrong things.

- Also see that you can't access the result when the publishDate is in the future or not set!

### Screenshots


Before:
![Bildschirmfoto von 2020-07-11 17-03-42](https://user-images.githubusercontent.com/57408107/87227078-96c29000-c398-11ea-9301-226ead9c7718.png)
After:
![Bildschirmfoto von 2020-07-14 13-42-42](https://user-images.githubusercontent.com/57408107/87421887-f5c21800-c5d7-11ea-9c3a-710407a1e528.png)
View result (same as course exercises):
![Bildschirmfoto von 2020-07-14 13-45-42](https://user-images.githubusercontent.com/57408107/87422158-6c5f1580-c5d8-11ea-94ba-be208ffbcd37.png)
